packages/node/src/indexer/fetch.module.ts

```diff
@@ -36,7 +38,20 @@ import { UnfinalizedBlocksService } from './unfinalizedBlocks.service';
     StoreCacheService,
     ApiService,
     IndexerManager,
-    ConnectionPoolService,
+    ConnectionPoolStateManager,
+    {
+      provide: ConnectionPoolService,
+      useFactory: (
+        nodeConfig: NodeConfig,
+        poolStateManager: ConnectionPoolStateManager<ApiPromiseConnection>,
+      ) => {
+        return new ConnectionPoolService(
+          poolStateManager,
+          nodeConfig.batchSize,
```

Comment:
Maybe if the constructor takes NodeConfig then we can remove the need for a factory